src/lib.rs

```diff
@@ -1,3 +1,5 @@
+#![allow(clippy::needless_range_loop)]
```

Comment:
Why is this line needed?

Reply:
I wanted to separate this particular refactoring into a separate PR later, as one has to be extremely careful when refactoring there:
```rust
src/bucket_msm.rs|99 col 20 warning| the loop variable `win` is used to index `normal_slices`
||    |
|| 99 |         for win in 0..normal_slices.len() {
||    |                    ^^^^^^^^^^^^^^^^^^^^^^
||    |
||    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_range_loop
||    = note: `#[warn(clippy::needless_range_loop)]` on by default
|| help: consider using an iterator and enumerate()
||    |
|| 99 |         for (win, <item>) in normal_slices.iter().enumerate() {
||    |             ~~~~~~~~~~~~~    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/bucket_msm.rs|109 col 20 warning| the loop variable `win` is used to index `normal_slices`
||     |
|| 109 |         for win in 0..normal_slices.len() {
||     |                    ^^^^^^^^^^^^^^^^^^^^^^
||     |
||     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_range_loop
|| help: consider using an iterator and enumerate()
||     |
|| 109 |         for (win, <item>) in normal_slices.iter().enumerate() {
||     |             ~~~~~~~~~~~~~    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/bucket_msm.rs|130 col 20 warning| the loop variable `win` is used to index `phi_slices`
||     |
|| 130 |         for win in 0..phi_slices.len() {
||     |                    ^^^^^^^^^^^^^^^^^^^
||     |
||     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_range_loop
|| help: consider using an iterator and enumerate()
||     |
|| 130 |         for (win, <item>) in phi_slices.iter().enumerate() {
||     |             ~~~~~~~~~~~~~    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/bucket_msm.rs|140 col 20 warning| the loop variable `win` is used to index `phi_slices`
||     |
|| 140 |         for win in 0..phi_slices.len() {
||     |                    ^^^^^^^^^^^^^^^^^^^
||     |
||     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_range_loop
|| help: consider using an iterator and enumerate()
||     |
|| 140 |         for (win, <item>) in phi_slices.iter().enumerate() {
||     |             ~~~~~~~~~~~~~    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
src/bucket_msm.rs|160 col 20 warning| the loop variable `win` is used to index `slices`
||     |
|| 160 |         for win in 0..slices.len() {
||     |                    ^^^^^^^^^^^^^^^
||     |
||     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_range_loop
|| help: consider using an iterator and enumerate()
||     |
|| 160 |         for (win, <item>) in slices.iter().enumerate() {
||     |             ~~~~~~~~~~~~~    ~~~~~~~~~~~~~~~~~~~~~~~~~
src/bucket_msm.rs|171 col 20 warning| the loop variable `win` is used to index `slices`
||     |
|| 171 |         for win in 0..slices.len() {
||     |                    ^^^^^^^^^^^^^^^
||     |
||     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_range_loop
|| help: consider using an iterator and enumerate()
||     |
|| 171 |         for (win, <item>) in slices.iter().enumerate() {
||     |             ~~~~~~~~~~~~~    ~~~~~~~~~~~~~~~~~~~~~~~~~
src/bucket_msm.rs|293 col 18 warning| the loop variable `i` is used to index `running_sums`
||     |
|| 293 |         for i in 1..running_sums.len() {
||     |                  ^^^^^^^^^^^^^^^^^^^^^
||     |
||     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_range_loop
|| help: consider using an iterator and enumerate()
||     |
|| 293 |         for (i, <item>) in running_sums.iter().enumerate().skip(1) {
||     |             ~~~~~~~~~~~    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
|| `ark-msm` (lib) generated 7 warnings
|| `ark-msm` (lib test) generated 7 warnings (7 duplicates)
```